sec_certs/dataset.py

```diff
@@ -650,34 +681,99 @@ def extract_data(self, fresh: bool = True, update_json: bool = False):
+        if not self.cve_dataset_path.exists() or download_fresh_cves is True:
+            cve_dataset = CVEDataset.from_web()
+            cve_dataset.to_json(str(self.cve_dataset_path))
+        else:
+            cve_dataset = CVEDataset.from_json(str(self.cve_dataset_path))
+
+        return cve_dataset
 
-        chunks = chunk_list(digests, constants.N_THREADS)
-        chunks_dicts = [{x: self[x].name for x in y} for y in chunks]
+    def compute_heuristics(self, update_json=True, download_fresh_cpes: bool = False):
+        def compute_candidate_versions():
+            logger.info('Computing heuristics: possible product versions in certificate name')
+            for cert in self:
+                cert.get_heuristics_version()
 
-        results = cert_processing.process_parallel(helpers.match_certs, list(
-            zip(chunks_dicts, [cpe_titles for _ in range(constants.N_THREADS)])), constants.N_THREADS,
-                                                   use_threading=False, unpack=True)
+        def compute_cpe_matches(cpe_dataset: CPEDataset):
+            logger.info('Computing heuristics: Finding CPE matches for certificates')
+            for cert in self:
+                cert.get_heuristics_cpe_match(cpe_dataset)
 
-        for chunk in results:
-            for digest, matches in chunk.items():
-                self[digest].cpe_matching = matches
+        compute_candidate_versions()
+        cpe_dset = self.prepare_cpe_dataset(download_fresh_cpes)
+        compute_cpe_matches(cpe_dset)
 
         if update_json is True:
             self.to_json(self.json_path)
 
+    def manually_verify_cpe_matches(self, update_json=True):
+        def verify_certs(certificates_to_verify: List[CommonCriteriaCert]):
+            n_certs_to_verify = len(certificates_to_verify)
+            for i, x in enumerate(certificates_to_verify):
+                print(f'\n[{i}/{n_certs_to_verify}] Vendor: {x.manufacturer}, Name: {x.name}')
+                for index, c in enumerate(x.heuristics.cpe_matches):
+                    print(f'\t- {[index]}: {c[1]}')
+                print(f'\t- [A]: All are fitting')
+                print(f'\t- [X]: No fitting match')
+                inpts = input('Select fitting CPE matches (split with comma if choosing more):').strip().split(',')
+
+                if 'X' not in inpts and 'x' not in inpts:
+                    if 'A' in inpts or 'a' in inpts:
+                        inpts = [x for x in range(0, len(x.heuristics.cpe_matches))]
+                    try:
+                        inpts = [int(x) for x in inpts]
+                        if min(inpts) < 0 or max(inpts) > len(x.heuristics.cpe_matches) - 1:
+                            raise ValueError(f'Incorrect number chosen, choose in range 0-{len(x.heuristics.cpe_matches) - 1}')
+                    except ValueError as e:
+                        logger.error(f'Bad input from user, repeating instance: {e}')
```

Comment:
Is using `logger` to communicate with the user here a good idea? What if `logger` is redirected somewhere? Logs of the library and user printed output do not mix well in this case. I think just prints or perhaps the use of click here is ok.